### DISCUSSION TITLE
Combine `PresetDisplacement` with `DirichletBC`

### DISCUSSION THREAD

**Discussion by wehrad** [idaholab/moose#20574] [Q&A Getting Started] [answered]
Hi everyone,

in a [MASTODON ](https://github.com/idaholab/mastodon) application for some simple elastic deformation,  I would like to apply both a `DirichletBC` and a `PresetDisplacement` (as e.g [here](https://github.com/idaholab/mastodon/blob/devel/examples/ex14/building_basemat_with_isolators_new.i#L495)) simultaneously on the same side of my domain as the process I am studying is a combination of direct ground shaking and pressure release. 

However, my `DirichletBC` gets disabled as soon as I enable my `PresetDisplacement` BC, and that even out of the period where the preset displacement is defined.

I tried to find examples that would guide me but I couldn't find any so far, so I start wondering if such a situation is even possible/makes sense? 

Thank you a lot in advance for your thoughts and help!

**Comment by GiudGiud** [comment]
@cbolisetti any ideas on this?

**Comment by cbolisetti** [comment, marked as answer]
HI @AdrienWehrle are you trying to apply a `DirichletBC` and `PresetDisplacement` simultaneously to different dofs? In the MASTODON example that you referenced the BCs are applied to different dofs and that should work just fine. I don't think you can apply them simultaneously to the same dof. Also, if you can share your input file, that would be helpful.

**Reply by wehrad** [reply]
Hi @cbolisetti! 
Thank you a lot for your help! I'm trying to constrain the displacement in a given direction with a Dirichlet BC + a variable pressure BC. Both result in a displacement (directly or as a result of, respectively) and I would like to combine them on the same node. In the example I refer to, the only BC for, e.g., `accel_x` is `x_motion`, it is not combined with a Dirichlet BC on the same variable. Again, I am not sure this is actually possible/makes sense.
https://github.com/idaholab/mastodon/blob/f11984c9b64481322afff56448d678f19d646cc5/examples/ex14/building_basemat_with_isolators_new.i#L478

I will share an input file shortly but the idea would be to have the following BCs on one of the sides of a rectangular prism:

```
[BCs]
  [Pressure]
    [side_pressure]  
    boundary = side_1
    function = pressure_release
    displacements = 'disp_x disp_y disp_z'
    []
  []

  [tremor]
    type = PresetDisplacement
    boundary = 'side_1'
    function = disp_BC
    variable = 'disp_x'
    beta = 0.25
    acceleration = 'accel_x'
    velocity = 'vel_x'
  []
[]
```

with functions being e.g.: 

```
[Functions]
  [pressure_release]
    type = ParsedFunction
    value = 'if (t < 0.1, 8829*(1000-z), 8829*(1000-z) - 1*8829*(1000))'   
  []
  [disp_BC]
    type = PiecewiseLinear
    data_file = Ormsby.csv
    format = 'columns'
  []
[]
```

The problem is that I don't obtain a `disp_x` which is a combination of the two BCs, only `side_pressure` is applied.

Just because that might help to understand why I want to do that: the science case behind is to combine a pressure release + tremor (the Ormsby wavelet is used here) on the same side of the domain.

Thank you a lot for your help!

**Reply by cbolisetti** [reply]
Hi @AdrienWehrle, I see what you mean now, thanks for elaborating. PressureBC is an integrated BC and PresetDisplacement is a dirichlet BC, so we should be able to apply them together. I am not sure why it wouldn't work. Perhaps @bwspenc or @GiudGiud would know?

**Reply by cbolisetti** [reply]
Actually, on further thought, I don't applying a pressure and a Dirichlet BC on the same boundary would make a lot of sense. If the DirichletBC is applied, the `disp_x` is basically prescribed and any other BC on the same boundary and dof will not make any difference. Perhaps that is why Dirichlet BCs are turned off when you apply them together?

**Reply by wehrad** [reply]
> Perhaps that is why Dirichlet BCs are turned off when you apply them together?

Hi @cbolisetti, that is indeed also my guess... Do you see any other possibility to combine two forcings on the same boundary? Maybe should I combine them in pre-processing and feed the result into a PresetDisplacement... But the tricky part is my pressure release depends on elevation and is therefore not constant across the whole boundary like a PresetDisplacement... 
Thank you a lot for your help!
